packages/shared/src/components/modals/CommentBox.tsx

```diff
@@ -13,13 +14,19 @@ import styles from './CommentBox.module.css';
 import { ProfilePicture } from '../ProfilePicture';
 import Markdown from '../Markdown';
 import { RecommendedMentionTooltip } from '../tooltips/RecommendedMentionTooltip';
-import {
-  fixHeight,
-  UPDOWN_ARROW_KEYS,
-  UseUserMentionOptions,
-} from '../../hooks/useUserMention';
+import { fixHeight, UseUserMentionOptions } from '../../hooks/useUserMention';
 import { Post } from '../../graphql/posts';
 import { cleanupEmptySpaces } from '../../lib/strings';
+import {
+  ArrowKey,
+  BaseInputEvent,
+  checkIsKeyboardCommand,
+  KeyboardCommand,
+  Y_AXIS_KEYS,
+} from '../../lib/element';
+
+type TextareaInputEvent = CompositionEvent<HTMLTextAreaElement> &
+  BaseInputEvent;
```

Comment:
I think you have this as TextareaEvent right?
Perhaps simply export that one? *(lib/element)

Reply:
The event we can also use is `KeyboardEvent<HTMLTextAreaElement>` - but for the `oninput` event, the type that has the `data` property when it triggers is the `CompositionEvent` which is paired to the `HTMLTextAreaElement`.